bpf/lib/wireguard.h

```diff
@@ -133,6 +133,12 @@ wg_maybe_redirect_to_encrypt(struct __ctx_buff *ctx)
 	if (!src || !identity_is_cluster(src->sec_identity))
 		goto out;
 
+	/* If source is remote node we should treat it like outside traffic.
+	 * This is possible when connection is done from pod to load balancer with DSR enabled.
+	 */
+	if (identity_is_remote_node(src->sec_identity))
```

Comment:
I'm assuming this is fine with node-to-node encryption too. Since the remote node IP is only in the `allowedIPs` list of the peer representing the remote node, we cannot forward such packets as they will have the wrong source IP. Is that correct?

Reply:
Yes. It's correct.